### PR TITLE
feat(management): avoid self-referencing JMX operations

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes https://github.com/cryostatio/cryostat-agent/issues/278
Depends on https://github.com/cryostatio/cryostat-core/pull/309
Depends on https://github.com/cryostatio/cryostat-core/pull/313
Depends on #289
Depends on #293

~~TODO before merge: correct `-core` version in `pom.xml` to point to a newly-released upstream version, not the `-SNAPSHOT` as it is for testing purposes in this PR~~

### CODE REVIEW

File: src/main/java/io/cryostat/agent/remote/RecordingsContext.java [hunk 1]
```diff
@@ -231,146 +214,112 @@ private void handleStartRecordingOrSnapshot(HttpExchange exchange) throws IOExce
...
             log.error("Failed to start recording", e);
             exchange.sendResponseHeaders(HttpStatus.SC_INTERNAL_SERVER_ERROR, BODY_LENGTH_NONE);
         } catch (InvalidEventTemplateException e) {
+            log.warn("Invalid custom event template", e);
             exchange.sendResponseHeaders(HttpStatus.SC_BAD_REQUEST, BODY_LENGTH_NONE);
         }
     }
 
-    private void handleStopOrUpdate(HttpExchange exchange, long id) throws IOException {
+    private void handleStopOrUpdate(HttpExchange exchange, long recordingId) throws IOException {
```
Comment (mwangggg) [inline src/main/java/io/cryostat/agent/remote/RecordingsContext.java:226]: Is there a reason `id` was changed to `recordingId` here? I noticed it wasn't in `handleGetRecording` and `handleDelete`

File: src/main/java/io/cryostat/agent/MainModule.java [hunk 2]
```diff
@@ -242,76 +236,14 @@ public static FlightRecorderHelper provideFlightRecorderHelper() {
...
-        }
-    }
-
-    @Provides
-    @Singleton
-    public static Environment provideEnvironment(@Named(TEMPLATES_PATH) Path templatesTmp) {
-        return new Environment() {
-            @Override
-            public String getEnv(String key) {
-                if (LocalStorageTemplateService.TEMPLATE_PATH.equals(key)) {
```
Comment (mwangggg) [inline src/main/java/io/cryostat/agent/MainModule.java:268]: Is there any more use for `TEMPLATE_PATH`? If not, it can be taken out at line 68

**Comment by mwangggg** [comment]
Hm I tried to start a recording and this happened...

**Comment by andrewazores** [comment]
Hmm. I'm not seeing that. Which target did you have selected? Did this only happen after you tried to expand the automated analysis?

**Comment by mwangggg** [comment]
> Hmm. I'm not seeing that. Which target did you have selected? Did this only happen after you tried to expand the automated analysis?

Oh I think I was using an out of date -core. I updated everything and reran and it is working as expected now :)

File: src/main/java/io/cryostat/agent/remote/RecordingsContext.java (hunk 1 above)
Comment (andrewazores) [reply src/main/java/io/cryostat/agent/remote/RecordingsContext.java:226]: I don't recall any particular reason, I guess this is just something I did while refactoring and cleaning up.

**Review by mwangggg — DISMISSED** [review]
lgtm

**Comment by andrewazores** [comment]
Rebased just to drop some accidentally included unrelated changes to the `maven-publish` CI workflow.

**Pull request merged**